pkg/kubernetes/accesscontrol_restmapper.go

```diff
@@ -0,0 +1,106 @@
+package kubernetes
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/restmapper"
+
+	"github.com/manusa/kubernetes-mcp-server/pkg/config"
+)
+
+type AccessControlRESTMapper struct {
+	delegate     *restmapper.DeferredDiscoveryRESTMapper
+	staticConfig *config.StaticConfig // TODO: maybe just store the denied resource slice
+}
+
+var _ meta.RESTMapper = &AccessControlRESTMapper{}
+
+// isAllowed checks the resource is in denied list or not.
+// If it is in denied list, this function returns false.
+func (a AccessControlRESTMapper) isAllowed(gvk *schema.GroupVersionKind) bool {
+	if a.staticConfig == nil {
```

Comment:
This logic is true. But if staticConfig is accidentally passed as nil, this would mean we allow all resources. Just brainstorming, would it be too disruptive if we return false, if staticConfig is nil? or can we return error in `NewAccessControlRESTMapper` if staticConfig is nil?. WDYT?

Reply:
Note: the method is the same as the original method, I just moved it here.

It's very likely that this method will be deeply refactored in the future.

Currently, `staticConfig` can be nil (in fact will be nil unless a config is explicitly loaded.

If we go with https://github.com/manusa/kubernetes-mcp-server/issues/151#issuecomment-3018150180 or similar, then probably staticConfig will always have a value and we can probably change this.